src/commands/createWebApp/readAzConfig.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as fse from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+import { IAppServiceWizardContext } from 'vscode-azureappservice';
+
+/**
+ * Takes any number of Azure config properties and returns their configured values
+ * found in `$AZURE_CONFIG_DIR/config`.
+ * Azure CLI configuration docs: https://aka.ms/AA64syh
+ */
+export async function readAzConfig(wizardContext: IAppServiceWizardContext, ...properties: AzConfigProperties[]): Promise<AzConfig> {
+    const config: AzConfig = {};
+    const configPath: string = path.join(os.homedir(), '.azure', 'config');
```

Comment:
The az cli actually provides a command to list out defaults. For example:
```
> az configure --list-defaults --output json
[
  {
    "name": "group",
    "source": "/Users/erijiz/.azure/config",
    "value": "myRG"
  },
  {
    "name": "location",
    "source": "/Users/erijiz/.azure/config",
    "value": "westus"
  }
]
```

First of all, this just looks easier to parse than all of your RegExp stuff. Second of all, it is more robust if the config isn't stored in the default manner. For example, the following command will add a config file in my current folder (similar to how VS Code has workspace settings instead of user settings):
```
> az configure --defaults web='emjtestdefault1' --scope local
> az configure --list-defaults --output json --scope local
[
  {
    "name": "web",
    "source": "/Users/erijiz/Repos/vscode-azurefunctions/.azure/config",
    "value": "emjtestdefault1"
  },
  {
    "name": "group",
    "source": "/Users/erijiz/.azure/config",
    "value": "myRG"
  },
  {
    "name": "location",
    "source": "/Users/erijiz/.azure/config",
    "value": "westus"
  }
]
```

Reply:
Is there a way to get the output of that command from within the extension?

Reply:
Yeah something like this: https://github.com/microsoft/vscode-azurefunctions/blob/master/src/utils/cpUtils.ts

If app service repo doesn't have any child_process utils yet, it's fine to just copy that file